controllers/product_controller.py

```diff
@@ -72,14 +84,98 @@ def get_product_categories():
 #Get Product List API
 def get_product_list():
     with engine.connect() as db:
-        query = select(product_table)
+        # Create a subquery that calculates the average rating for each product
+        subquery = select(review_table.c.product_id, func.round(func.avg(review_table.c.rating), 1).label("rating")).group_by(review_table.c.product_id).alias("subquery")
+        # Join the subquery with the product table to get the product details
+        query = select(product_table, subquery.c.rating).join(subquery, product_table.c.id == subquery.c.product_id, isouter=True)
         products = db.execute(query).fetchall()
+        # Convert the average rating from Decimal to float
         return APIHelper.send_success_response(data=[ProductModel(**row._mapping) for row in products])
 
 #Get Product details
 def get_product_details(product_id):
     with engine.connect() as db:
-        product = db.execute(select(product_table).where(product_table.c.id == product_id)).fetchone()
-        if not product:
+        query = select(
+            product_table.c.id,
+            product_table.c.name,
+            product_table.c.category,
+            product_table.c.price,
+            product_table.c.description,
+            product_table.c.discount,
+            func.round(func.avg(review_table.c.rating),1).label("average_rating"),
+            review_table.c.user_id,
+            users_table.c.name.label("user_name"),  # Add the user name
+            review_table.c.review,
+        ).select_from(
+            product_table
+        ).outerjoin(
+            review_table, product_table.c.id == review_table.c.product_id
+        ).outerjoin(
+            users_table, review_table.c.user_id == users_table.c.id  # Join with the users table
+        ).group_by(
+            product_table.c.id,
+            product_table.c.name,
+            product_table.c.category,
+            product_table.c.price,
+            product_table.c.description,
+            product_table.c.discount,
+            review_table.c.user_id,
+            users_table.c.name,  # Group by the user name
+            review_table.c.review,
+            review_table.c.rating
+        )
+        query = query.where(product_table.c.id == product_id)
```

Comment:
Do not give same variable name

Reply:
are u talking about the query i used at the start and with the where?

Reply:
Yes